qtwidgetsextra/groupbox/QAbstractButtonGroupBox.cpp

```diff
@@ -0,0 +1,341 @@
+{
+    Q_ASSERT(widget);
+    connect(group,SIGNAL(buttonClicked(QAbstractButton*)),this,SLOT(buttonClicked(QAbstractButton*)));
+    connect(group,SIGNAL(buttonPressed(QAbstractButton*)),this,SLOT(buttonPressed(QAbstractButton*)));
+    connect(group,SIGNAL(buttonReleased(QAbstractButton*)),this,SLOT(buttonReleased(QAbstractButton*)));
+    connect(group,SIGNAL(buttonToggled(QAbstractButton*,bool)),this,SLOT(buttonToggled(QAbstractButton*,bool)));
+}
+
+void QAbstractButtonGroupBoxPrivate::clear() const
+{
+    QList<QAbstractButton*> buttons = group->buttons();
+    foreach (QAbstractButton *button, buttons) {
+        mainLayout->removeWidget(button);
+        group->removeButton(button);
+        delete button;
+    }
+}
+
+void QAbstractButtonGroupBoxPrivate::setOrientation(Qt::Orientation newOrientation)
+{
+    if (orientation == newOrientation) {
+        return;
+    }
+
+    if (newOrientation == Qt::Vertical) {
+        mainLayout->setDirection(QBoxLayout::TopToBottom);
+    }
+    else {
+        mainLayout->setDirection(QBoxLayout::LeftToRight);
+    }
+
+    orientation = newOrientation;
+}
+
+QStringList QAbstractButtonGroupBoxPrivate::items() const
+{
+    QStringList list;
+    const QList<QAbstractButton*> buttons = group->buttons();
+    foreach (QAbstractButton *button, buttons) {
+
+        list.append(button->text());
+    }
+
+    return list;
+}
+
+void QAbstractButtonGroupBoxPrivate::removeItem(int index)
+{
+    Q_ASSERT(!indexOutOfRange(index));
+
+
+    QAbstractButton *button = qobject_cast<QAbstractButton*>(mainLayout->itemAt(index)->widget());
+    mainLayout->removeWidget(button);
+    group->removeButton(button);
+    delete button;
+}
+
+void QAbstractButtonGroupBoxPrivate::insertButton(int index, QAbstractButton *button, int buttonId)
+{
+    if (group->buttons().contains(button)) {
+        return;
+    }
+
+    mainLayout->insertWidget(index,button);
+    group->addButton(button,buttonId);
+    button->setEnabled(enabled);
+}
+
+void QAbstractButtonGroupBoxPrivate::addButton(QAbstractButton *button, int buttonId)
+{
+    insertButton(mainLayout->count(),button, buttonId);
+}
+
+void QAbstractButtonGroupBoxPrivate::setCheckable(bool isCheckable)
+{
+    checkable = isCheckable;
+    const QList<QAbstractButton*> buttons = group->buttons();
+    foreach (QAbstractButton *button, buttons) {
+        button->setCheckable(checkable);
+    }
+}
+
+void QAbstractButtonGroupBoxPrivate::setChecked(bool checked, int index)
+{
+    Q_ASSERT(!indexOutOfRange(index));
+
+    QAbstractButton *button = qobject_cast<QAbstractButton*>(mainLayout->itemAt(index)->widget());
+    button->setChecked(checked);
+}
+
+bool QAbstractButtonGroupBoxPrivate::isChecked(int index)
+{
+    Q_ASSERT(!indexOutOfRange(index));
+
+    const QAbstractButton *button = qobject_cast<QAbstractButton*>(mainLayout->itemAt(index)->widget());
+    return button->isChecked();
+}
+
+void QAbstractButtonGroupBoxPrivate::setEnabled(bool enable)
```

Comment:
Hum depends on what we want here. If we use the enable of groupbox, the title of the groupbox will be disable to. Hum i've no preference here, as you want.